crates/client/rpc/src/lib.rs

```diff
@@ -937,3 +962,47 @@ where
         }
     }
 }
+
+/// Helper function to get filter events using address and keys
+///
+/// # Arguments
+///
+/// * `events` - A vector of all events
+/// * `address` - Address to use to filter the events
+/// * `keys` - Keys to use to filter the events. An event is filtered if any key is present
+/// * `max_results` - Optional, indicated the max events that need to be filtered
+///
+/// # Returns
+///
+/// * `(block_events: Vec<EventWrapper>, continuation_token: usize)` - A tuple of the filtered
+///   events and the first index which still hasn't been processed block_id and an instance of Block
+pub fn filter_events_by_params(
```

Comment:
move it in a different files `events.rs`
Along with the `get_block_events` method and basically anything event-related you can find in the file.

It will help readablility and code maintaining

Reply:
moved `filter_events_by_params` to a new file, however `get_block_events` depends on `self`, so should that stay inside the implementation

Reply:
you can have multiple `impl MyType {}` blocks

Reply:
moved to `events.rs`